src/pynxtools_xps/kratos/metadata_kratos.py

```diff
@@ -0,0 +1,350 @@
+        # header, data = self._separate_header_and_data()
+        # self.parse_header_into_metadata(header)
+
+        return self.spectra
+
+    def parse_header_into_metadata(self, header: List[str]):
+        """
+        Parse header into KratosMetadata dataclass.
+
+        Parameters
+        ----------
+        header : str
+            Header data for one spectrum as a String.
+
+        """
+        datacls_fields = list(self.metadata.__dataclass_fields__.keys())
+        datacls_fields = [field for field in datacls_fields if "_units" not in field]
+
+        for line in header:
+            try:
+                key, value = line.split(" : ")
+
+            except ValueError:
+                continue
+
+            key = convert_pascal_to_snake(key)
+            key = SETTINGS_MAP.get(key, key)
+
+            if key in datacls_fields:
+                field_type = type(getattr(self.metadata, key))
+
+                if key in KEYS_WITH_UNITS:
+                    value, unit = self.extract_unit(key, value)
+                    setattr(self.metadata, f"{key}_units", unit)
+
+                value = self.map_values(key, value, field_type)
+
+                setattr(self.metadata, key, value)
+
+        self.metadata.validate_types()
+
+    def extract_unit(self, key: str, value):
+        """
+        Extract units for the metadata containing unit information.
+
+        Example:
+            analyser_work_function: 4.506eV
+            -> analyser_work_function: 4.506,
+               analyser_work_function_units: eV,
+
+        Parameters
+        ----------
+        key : str
+            Key of the associated value.
+        value : str
+            Combined unit and value information.
+
+        Returns
+        -------
+        value :
+            value with units.
+        unit : str
+            Associated unit.
+
+        """
+
+        pattern = re.compile(r"([-+]?\d*\.?\d+(?:[eE][-+]?\d+)?)([a-zA-Z]+)")
+        match = pattern.match(value)
+
+        if match:
+            value, unit = match.groups()
+        else:
+            unit = ""
+
+        unit = convert_units(unit)
+
+        if key in UNIT_MISSING:
+            unit = UNIT_MISSING[key]
+
+        return value, unit
+
+    def map_values(self, key: str, value, field_type):
+        """
+        Map values to corresponding structure and field type.
+
+        Parameters
+        ----------
+        key : str
+            KratosMetadata dataclass key.
+        value : str
+            Original value.
+        field_type : class
+            Class to be mapped onto.
+
+        Returns
+        -------
+        value
+            Value of correct type and internal structure.
+
+        """
+
+        value_function_map: Dict[str, Any] = {
+            "date_created": _parse_datetime,
+            "description": _convert_description,
+            "charge_neutraliser": convert_bool,
+            "deflection": _convert_xray_deflection,
+        }
```

Comment:
you can move it on the class leve. Because for every key  the same dict is being created adding unnecessary variable creation time slowing the reader.

Reply:
moved, also for the Phi reader